contracts/ERC721M.sol

```diff
@@ -197,7 +207,32 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable {
         bytes32[] calldata proof,
         uint256 timestamp,
         bytes calldata signature
-    ) external payable canMint hasSupply(qty) {
+    ) external payable {
+        _mintInternal(qty, msg.sender, proof, timestamp, signature);
+    }
+
+    function crossmint(
+        uint32 qty,
+        address to,
+        bytes32[] calldata proof,
+        uint256 timestamp,
+        bytes calldata signature
+    ) external payable {
+        if (_crossmintAddress == address(0)) revert CrossmintAddressNotSet();
+
+        // Check the caller is Crossmint
+        if (msg.sender != _crossmintAddress) revert CrossmintOnly();
+
+        _mintInternal(qty, to, proof, timestamp, signature);
+    }
+
+    function _mintInternal(
+        uint32 qty,
+        address to,
+        bytes32[] calldata proof,
+        uint256 timestamp,
+        bytes calldata signature
+    ) internal canMint hasSupply(qty) {
```

Comment:
assertValidCosign(msg.sender, qty, timestamp, signature); for this line, I'm thinking do we need to change this to `to`

Reply:
yeah, second thought is that yeah, it should be msg.sender, that we cosign to stamp for the msg sender, not on behalf of the `to`, even though for the `mint()` that msg.sender == to.

Reply:
this means that if we turn on cosign, crossmint also need to talk to our api to get the signature first. @channing-magiceden let's chat with crossmint to see if they can support it, otherwise we might need to make this cosign check optional